pkg/controller/machine.go

```diff
@@ -245,16 +258,41 @@ func (c *controller) addNodeToMachine(obj interface{}) {
 		return
 	}
 
-	klog.V(4).Infof("Add machine object backing node %q", machine.Name)
-	c.enqueueMachine(machine)
+	if machine.Status.CurrentStatus.Phase == v1alpha1.MachineCrashLoopBackOff {
+		return
+	} else if nodeConditionsHaveChanged(machine.Status.Conditions, node.Status.Conditions) {
```

Comment:
```suggestion
	if machine.Status.CurrentStatus.Phase != v1alpha1.MachineCrashLoopBackOff && nodeConditionsHaveChanged(machine.Status.Conditions, node.Status.Conditions) {
```
To keep the code more readable.